proposals/NNNN-developer-oem-communication.md

```diff
@@ -0,0 +1,153 @@
+* Initiate, view, and reply to general text-based messages from an application developer
+* Open issues against an app - each with a title, description, and whether or not the issue should be public to other OEMs
+* See all open and closed issues for apps which were either created by the OEM or made public by another OEM
+* Close self-authored issues that have been deemed solved or irrelevant
+* Re-open self-authored closed issues
+* Exchange text-based messages with the app developer regarding a specific issue
+
+App developers should be able to:
+* Initiate, view, and reply to general text-based messages from all OEMs
+* See all open and closed issues against their app
+* Exchange text-based messages with the OEM regarding a specific issue
+
+SDLC Administrators should be able to:
+* See the opened and closed issues for each app, but not the messages exchanged between OEMs and app developers
+
+## Proposed solution
+
+### Overview
+The proposed solution is to create an administrative enhancement to SHAID to support the creation, retrieval, updating, and deletion (as appropriate) of text-based messages and issues between an OEM and app developer with the behavior defined above.
+
+This would require the following new APIs which would be utilized by the Developer Portal:
+
+* POST `/application/message`: creates a new text-based message for an application-level thread
+* GET `/application/message`: retrieves text-based messages for an application-level thread
+
+* POST `/application/issue`: creates a new issue for an application
+* GET `/application/issue`: retrieves issues for an application
+* PUT `/application/issue`: update an issue as closed or open
+
+* POST `/application/issue/message`: creates a new text-based discussion message for a specific issue
+* GET `/application/issue/message`: retrieves text-based messages for a specific issue
+
+And modifications for existing APIs as follows:
+
+* GET `/application`: to retrieve applications with a negative OEM allowance filter (that is, retrieve SDLC-certified applications which have not been granted access to a specific OEM by the app developer)
+* GET `/vendor`: to retrieve vendors with a negative OEM allowance filter ()
+
+### Data Storage
+The data for these new features would be stored in new relational tables in the existing SHAID Postgres database, as follows:
+* `application_vendor_message`: stores an auto-generated ID, which application the message is regarding, which OEM thread the message is associated with, who the message is from (OEM or app developer ID), the name of the individual who authored the message, the message contents, creation time, deletion time
+* `application_issue`: stores a globally-unique auto-generated ID, an application-specific unique auto-generated ID, which application the issue is regarding, which OEM created the issue, the issue title, description, whether it is accessible by other OEMs, it's opened/closed status, creation time, updated time
```

Comment:
Update to: 
`application_issue`: stores a globally-unique auto-generated ID, an application-specific unique auto-generated ID, which application the issue is regarding, which OEM created the issue, the issue title, description, whether it is accessible by other OEMs, it's opened/closed status, **creation time/date and updated time/date**